### PR TITLE
AM-450:  Get case document metadata API at ccd-data-store.

### PR DESCRIPTION
**Pull request by SivaKanukolluQA**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-450
(Task: https://tools.hmcts.net/jira/browse/AM-550)

### Change description ###
Case Document AM API-Download: Implement 'Get case document metadata' API at ccd-data-store.

**Does this PR introduce a breaking change?** (check one with "x")
[ ] Yes
[x] No

### CODE REVIEW

File: src/aat/resources/features/F-1001/F-1001.feature [hunk 1]
```diff
@@ -0,0 +1,156 @@
+@F-1001
+Feature: F-1001: Get case document metadata for a given caseId and documentId
+
+  Background: Load test data for the scenario
+    Given an appropriate test context as detailed in the test data source
+
+  @S-1010
+  Scenario: must receive a document metadata and list of permissions for a valid Citizen user
+    Given a user with [an active Citizen profile in CCD with access to a document]
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-1001/F-1001.feature]: it reads like there is a user and a document in the context and the user has access to the doc. However, at this stage of the scenario, the document is not yet on CCD.

File: src/aat/resources/features/F-1001/S-1001.td.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    "an active Solicitor profile having full permissions on the same case type",
+    "has the case id just created where a document id is associated",
+    "contains the requested document am data"
+  ],
+
+  "expectedResponse": {
+    "_extends_": "Common_200_Response",
+    "body": {
+      "ResponseCode": "200",
+      "status": "OK",
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-1001/S-1001.td.json]: inherited from common 200 response already.

File: src/aat/resources/features/F-1001/S-1002.td.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      "status": "OK",
+      "timeStamp": "[[ANYTHING_PRESENT]]",
+      "ResponseBody": {
+        "caseDocumentMetadata": {
+          "caseId": "1234567890123456",
+          "caseTypeId": "123",
+          "jurisdictionId": "BEFTA2",
+          "documents": [
+            {
+              "url": "http://dm-store-aat.service.core-compute-aat.internal/documents/e56f0aea-4a5f-46d5-82d3-01b536420870",
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-1001/S-1002.td.json]: base url parameterised with env var

File: src/aat/resources/features/F-1001/S-1002.td.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      "ResponseCode": "200",
+      "status": "OK",
+      "timeStamp": "[[ANYTHING_PRESENT]]",
+      "ResponseBody": {
+        "caseDocumentMetadata": {
+          "caseId": "1234567890123456",
+          "caseTypeId": "123",
+          "jurisdictionId": "BEFTA2",
+          "documents": [
+            {
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-1001/S-1002.td.json]: What is this ResponseBody inside body, here?

File: src/aat/resources/features/F-1001/S-1001.td.json [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            {
+              "url": "http://dm-store-aat.service.core-compute-aat.internal/documents/e56f0aea-4a5f-46d5-82d3-01b536420870",
+              "name": "Screenshot 2019-11-14 at 1.57.32 pm.png",
+              "type": "png",
+              "description": "sample",
+              "id": "e56f0aea-4a5f-46d5-82d3-01b536420870",
+              "hasedtoken": null,
+              "permissions": "{C,R,U}"
+            }
+          ]
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-1001/S-1001.td.json]: Response body inside body?

File: src/aat/resources/features/F-1001/F-1001.feature [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+  @S-1008
+  Scenario: must return 401 when request provides expired authentication
+
+  @S-1009
+  Scenario: must return 401 when request provides expired s2s serviceAuthorisation
+
+  #Generic Scenarios for media type. No need to wrire for each feature
+  @S-1010
+  Scenario: must return 415 when request provides content type other than application/json
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-1001/F-1001.feature]: Pls @Ignore these.

Comment (SivaKanukolluQA) [reply src/aat/resources/features/F-1001/F-1001.feature]: Ignored these 3 scenarios.

File: src/aat/resources/features/F-1001/S-1001.td.json (hunk 5 above)
Comment (SivaKanukolluQA) [reply src/aat/resources/features/F-1001/S-1001.td.json]: Updated the response.
"expectedResponse": {
    "_extends_": "Common_200_Response",
    "body": {
      "_links": {
        "self": {
          "href": "http://localhost:4452/cases/1583170755994880/documents/e97f6615-e73a-4b43-a832-65ee5a2da7e9"
        }
      },
      "documentMetadata": {
        "caseId": "1583170755994880",
        "caseTypeId": "BEFTA_CASETYPE_2_1",
        "jurisdictionId": "BEFTA_JURISDICTION_2",
        "document": {
          "url": "http://dm-store:8080/documents/e97f6615-e73a-4b43-a832-65ee5a2da7e9",
          "name": "Screenshot 2020-02-17 at 12.51.46.png",
          "type": "Document",
          "description": null,
          "id": "e97f6615-e73a-4b43-a832-65ee5a2da7e9",
          "permissions": [
            "CREATE",
            "READ",
            "UPDATE"
          ]
        }
      }
    }
  }

File: src/aat/resources/features/F-1001/S-1002.td.json (hunk 3 above)
Comment (SivaKanukolluQA) [reply src/aat/resources/features/F-1001/S-1002.td.json]: Scenario modified as per latest update.

File: src/aat/resources/features/F-1001/S-1001.td.json (hunk 2 above)
Comment (SivaKanukolluQA) [reply src/aat/resources/features/F-1001/S-1001.td.json]: Updated.

File: src/aat/resources/features/F-1001/S-1002.td.json (hunk 4 above)
Comment (SivaKanukolluQA) [reply src/aat/resources/features/F-1001/S-1002.td.json]: Updated.

**Comment by aashishaggarwal123** [comment]
New branch AM-450_get_user_authorization_document_download_functional_test added on master. This duplicate branch and PR need to be closed.

**Pull request closed without merging**